docs/cli/plugin_implementation_guide.md

```diff
@@ -26,6 +26,8 @@ Current implementations:
 
 ### Repository
 
+NOTE: This is not applicable if context-aware plugin discovery is enabled within Tanzu CLI.
```

Comment:
Perhaps "context-aware plugin discovery" should be a link to one of the documentation sections from one of the files above?